File: march-2026/3-22-water-day/water-day.py
def leaky_pipe(volume, leak, hours, threshold):
  # for each passing hour
  for i in range(hours):
    # we calculate the new volume based on the formula
    volume = volume * (1 - leak / 100)

    # check if treshold
    if volume < threshold:
      # if volume less than threshold, it failed
      return -1

  # otherwise we return the new volume
  # once the loop is over, rounded by 2 decimals
  return round(volume, 2)

File: march-2026/3-22-water-day/water_day.js
// Water Day 💧
// shercodes

function leakyPipe(volume, leak, hours, threshold) {
  let currentVolume = volume; // Initial volume
  const lossRate = leak; // Percentage per hour
  const iterations = hours; // Number of hours
  let waterRemaining = 0; // Result
  
  for (let i = 0; i < iterations; i++) {
    currentVolume = currentVolume * [1 - (lossRate / 100)]; // Calculates the volume after each hour passed.
    if (currentVolume >= threshold) {
      waterRemaining = Number(currentVolume.toFixed(2));
    } else {
      waterRemaining = -1;
    }
  }

  return waterRemaining;
}